sync/src/block/extension.rs

```diff
@@ -999,59 +999,60 @@ impl Extension {
     }
 
     fn on_chunk_response(&mut self, from: &NodeId, roots: &[H256], chunks: &[Vec<u8>]) {
-        if let State::SnapshotChunk {
-            block,
-            ref mut restore,
-        } = self.state
-        {
-            for (r, c) in roots.iter().zip(chunks) {
-                if c.is_empty() {
-                    cdebug!(SYNC, "Peer {} sent empty response for chunk request {}", from, r);
+        let (block, restore) = match self.state {
+            State::SnapshotChunk {
+                block,
+                ref mut restore,
+            } => (block, restore),
+            _ => return,
+        };
+        for (r, c) in roots.iter().zip(chunks) {
```

Comment:
What happens if the size of `roots` and `chunks` are different?

Reply:
It won't happen since it is already checked in `is_valid_response`.

Reply:
If then, please add an assertion that the size of them is equal.

Reply:
Added `assert_eq` here.